### PR TITLE
Updating Browser Window Documentation

### PR DESCRIPTION
**Pull request by jlord**
I've pulled `BrowserWindow` documentation work into its own PR since it's such a large and important part of the docs. I've done some things and I've got some questions!

- To reduce length, have pulled `webContents` and `session` into their own document :ok: :grey_question: 
- I link to the new `webContents` and `session` from their places in `BrowserWindow` doc.
- In most places I've gotten rid of a level of list nesting so that we don't hit the short limit from our markdown parser.
- In `BrowserWindow` the instance methods were noted as `BrowserWindow.devToolsWebContents` which I feel is a bit misleading since you can't call `devToolsWebContents` _on_ `BroswerWindow` but instead need an instance of it. Also confusing since the prototype methods are written in this same way. So I've given an example of `var win = new BrowserWindow()` and then made the method headings include `win.` How do people feel about this? Screenshot below. :ok: :grey_question: 

screen shot 2015-08-20 at 3 27 31 pm

- Consistently kept `session` and `webContents` with lowercase first letter.
- Applied other standardization mentioned [here](https://github.com/atom/electron/pull/2533#issue-102116862).
- Am trying to figure out the best way to document `session` and `webContents` methods which are properties on properties... I add examples showing how to access each object but keep the method headers `session.foo`. Example of the how to access code snippet:

```javascript
var BrowserWindow = require('browser-window');

var win = new BroswerWindow({ width: 800, height: 600 });
win.loadUrl("http://github.com");

var session = win.webContents.session
```

### CODE REVIEW

**Comment by OlsonDev** [comment]
Regarding `win.webContents` vs. `BrowserWindow.webContents`, I think that's a fine change but some other considerations are the jQuery way of documenting things: `.webContents` or the Prototypejs way: `BrowserWindow#webContents`. I tend to like the Prototype way coupled with an actual example (for novices who don't realize `#` means "prototype of". *Shrugs*

**Comment by OlsonDev** [comment]
Okay, I'm done commenting on stuff you probably had nothing to do with originally. :-)

File: docs/api/browser-window.md [hunk 1]
```diff
@@ -228,7 +235,7 @@ Emitted when devtools is focused / opened.
 
 ### Event: 'app-command':
 
-Emitted when an [App Command](https://msdn.microsoft.com/en-us/library/windows/desktop/ms646275(v=vs.85).aspx) is invoked. These are typically related to keyboard media keys or browser commands, as well as the "Back" button built into some mice on Windows.
+Emitted when an [App Command](https://msdn.microsoft.com/en-us/library/windows/desktop/ms646275(v=vs.85).aspx is invoked. These are typically related to keyboard media keys or browser commands, as well as the "Back" button built into some mice on Windows.
```
Comment (hokein) [inline docs/api/browser-window.md]: Missing a `)` after `https://msdn.microsoft.com/.../.aspx`.

**Comment by jlord** [comment]
Ok, this branch has been saved from the submodule/commit revert disaster I made of it yesterday and is ready for :eyes: :tada:

**Comment by thomasjo** [comment]
Wow, awesome job :clap: 

Looks good to me! :ship:

**Comment by jlord** [comment]
Awesome! :pizza:

**Pull request merged**